### PR TITLE
Actively look for a plugin that failed to start.

### PR DESCRIPTION
**Pull request by kohsuke**
In particular, I had a situation where a plugin under test failed to start, which isn't obvious to diagnose in this case

@reviewbybees

### CODE REVIEW

**Comment by stephenc** [comment]
:bee:

File: src/main/java/org/jvnet/hudson/test/PluginAutomaticTestBuilder.java [hunk 1]
```diff
@@ -86,8 +87,13 @@ public void testCliSanity() {
...
             String plugin = (String) params.get("artifactId");
             if (plugin != null) {
+                // did any plugin fail to start?
+                for (FailedPlugin fp : Jenkins.getInstance().getPluginManager().getFailedPlugins()) {
+                    throw new Error("Plugin "+fp.name+" failed to start", fp.cause);
+                }
+
                 PluginWrapper pw = Jenkins.getInstance().getPluginManager().getPlugin(plugin);
-                assertNotNull(pw);
+                assertNotNull(plugin+" failed to start", pw);
```
Comment (recena) [inline src/main/java/org/jvnet/hudson/test/PluginAutomaticTestBuilder.java]: `assertNotNull(plugin + " failed to start", pw);` in order to follow the same convention below

**Comment by recena** [comment]
:bee:

**Comment by oleg-nenashev** [comment]
:bee:

**Pull request merged**